### PR TITLE
feat: Redesign reset password page

### PR DESCRIPTION
**Pull request by bistaastha**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=lxw6nnM)
## Clickup
https://app.clickup.com/t/86cx2t8af

## UI Preview
Screenshot 2024-12-09 at 6 01 01 AM
Screenshot 2024-12-09 at 6 01 24 AM
Screenshot 2024-12-09 at 6 12 17 AM

## Summary by CodeRabbit

- **New Features**
	- Enhanced password reset form with improved layout and validation.
	- Added snack bar notifications for password change success and error messages.
	- New navigation options, including a back button to redirect to the sign-in page.

- **Bug Fixes**
	- Resolved issues with password validation logic for better user experience.

- **Style**
	- Updated CSS structure for a more organized and modular approach to styling.

- **Tests**
	- Adjusted test suite to reflect changes in password validation methods and user feedback mechanisms.

### CODE REVIEW

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/auth/new-password/new-password.page.ts [hunk 1]
```diff
@@ -96,36 +76,57 @@ export class NewPasswordPage implements OnInit {
...
+            message: 'Something went wrong. Please try after some time.',
+          };
+
+          this.matSnackBar.openFromComponent(ToastMessageComponent, {
+            ...this.snackbarPropertiesService.setSnackbarProperties('failure', toastMessageData),
+            panelClass: ['msb-failure'],
           });
-
-          await popup.present();
+          this.router.navigate(['/', 'auth', 'sign_in']);
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/new-password/new-password.page.ts:99]: Retaining current error messages

File: src/app/post-verification/invited-user/invited-user.page.spec.ts [hunk 2]
```diff
@@ -109,19 +109,19 @@ describe('InvitedUserPage', () => {
...
     });
 
     it('should return an error object when isPasswordValid is false', () => {
       component.isPasswordValid = false;
 
-      const result = component.customPasswordValidator();
+      const result = component.checkPasswordValidity();
 
       expect(result).toEqual({ invalidPassword: true }); // Error object
     });
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.spec.ts:127]: isPasswordValid will never be undefined

File: src/app/auth/new-password/new-password.page.ts [hunk 3]
```diff
@@ -30,62 +31,41 @@
...
-          Validators.minLength(12),
-          Validators.maxLength(32),
-          Validators.pattern(/[A-Z]/),
-          Validators.pattern(/[a-z]/),
-          Validators.pattern(/[0-9]/),
-          Validators.pattern(/[!@#$%^&*()+\-:;<=>{}|~?]/),
-        ]),
-      ],
+      password: ['', [Validators.required, this.checkPasswordValidity]],
+      confirmPassword: ['', [Validators.required, this.validatePasswordEquality]],
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/new-password/new-password.page.ts:59]: not used anywhere in the app

File: src/app/auth/new-password/new-password.page.scss [hunk 4]
```diff
@@ -1,84 +1,111 @@
-$new-password-header: #220033;
-$form-header: #000;
-$form-subheader: #4a4a4a;
-$password-icon: #b9beba;
-$secondary-cta-border: #e0e0e0;
-$flash-yellow: #fde081;
-$error: #f00;
-$offline-title: #4a4a4a;
-$offline-sub-title: #ababab;
+@import '../../../theme/colors.scss';
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/auth/new-password/new-password.page.scss:2]: not needed

**Pull request merged**